src/MultiTypeSymbolTable.chpl

```diff
@@ -320,19 +320,36 @@ module MultiTypeSymbolTable
         */
         proc info(name:string): string throws {
             var s: string;
-            if name == "__AllSymbols__" {
+            if tab.size == 0 {
+                s = "the symbol table is empty";
+            }
+            else if name == "__AllSymbols__" {
                 for n in tab {
-                    s += "name:%t dtype:%t size:%t ndim:%t shape:%t itemsize:%t\n".format(n, 
+                    s += "name:%t dtype:%t size:%t ndim:%t shape:%t itemsize:%t registered:%t\n".format(n, 
                               dtype2str(tab.getBorrowed(n).dtype), tab.getBorrowed(n).size, 
                               tab.getBorrowed(n).ndim, tab.getBorrowed(n).shape, 
-                              tab.getBorrowed(n).itemsize);
+                              tab.getBorrowed(n).itemsize, registry.contains(n));
                 }
-            } else {
+            } 
+            else if name == "__RegisteredSymbols__" {
+                if registry.size == 0 {
+                    s = "the registry is empty";
+                }
+                else {
+                    for n in registry {
+                        s += "name:%t dtype:%t size:%t ndim:%t shape:%t itemsize:%t registered:%t\n".format(n, 
+                                  dtype2str(tab.getBorrowed(n).dtype), tab.getBorrowed(n).size, 
+                                  tab.getBorrowed(n).ndim, tab.getBorrowed(n).shape, 
+                                  tab.getBorrowed(n).itemsize, registry.contains(n));
```

Comment:
I think you did the right thing following suit from the previous code.  Two future optimizations we could probably make here are:
 1. Pull the item out of the map (`tab`) once vs the multiple `tab.getBorrowed(n)` calls.
 2. It looks like we're generating/formatting the same underlying string for AllSymbols, RegisteredSymbols, and single items.  If that's the case it might be worth it to consolidate that into its own function and just call it so it ends up looking something like:
 ```
for n in (registry|tab|etc.) {
  s += formatEntry(n, tab.getBorrowed(n));
}
...
proc formatEntry(name:string, item:GenSymEntry): string {
  return "name:%t dtype:%t size:%t ...".format(name, dtype2str(item.dtype), item.size ...);
}
```

If the strings are not in fact the same, forget I said anything :)
(No change requested from me.)